### PR TITLE
This is the first Grammar file for syntax highlighting in the VS Code editor for the language P. I grouped together all the keywords as well as highlighting within tuples and attribute accesses. 

### PR DESCRIPTION
**Pull request by suesther**
Instructions for using Syntax Highlighting:
1. Move this folder into the ~/.vscode/extensions directory.
2. Add {"identifier":{"id":"undefined_publisher.peasy-ide-vscode"},"version":"0.0.1","location":{"$mid":1,"path":"~/.vscode/extensions/peasy-ide-vscode","scheme":"file"},"relativeLocation":"peasy-ide-vscode"} to ~/.vscode/extensions/extensions.json
3. Restart VS Code.
4. Feel free to change to different themes, or follow the instructions below to use the P Team's custom theme!

Instructions for using P's Custom Theme File:
1. Move the p-theme file inside the folder into the ~/.vscode/extensions directory.
2. Add {"identifier":{"id":"undefined_publisher.p-theme"},"version":"0.0.1","location":{"$mid":1,"path":"~/.vscode/extensions/p-theme","scheme":"file"},"relativeLocation":"p-theme"} to ~/.vscode/extensions/extensions.json
3. Restart VS Code. 
4. Change the theme in VS Code: Code -> Settings -> Theme -> Dropdown Menu (Select P's Custom Theme).

Section 1: Summary - Purpose of the change and the context (What, why and how)
This is the first Grammar file for syntax highlighting in the VS Code editor for the language P. I grouped together all the keywords as well as highlighting within tuples and attribute accesses. The screenshots below show how the grammar file works to categorize all the keywords and tokens in a sample P file. My workflow involved writing a yaml file, then using the command 
Screenshot 2023-06-06 at 1 33 24 PM
in order to convert the yaml file into a json file. This json file is what is being used by the extension to actually categorize all the keywords and tokens. 
This is what syntax highlighting looks like in the provided theme "Dark Modern."
Screenshot 2023-06-06 at 2 17 07 PM
Screenshot 2023-06-06 at 2 18 30 PM

This is what syntax highlighting looks like using the P theme file. 
Screenshot 2023-06-06 at 2 19 41 PM
Screenshot 2023-06-06 at 2 20 02 PM

Section 2: Testing
My testing was done within the VS Code terminal. By pressing F5 inside the folder inside VS Code, I could open a sample file in P where I could see the changes I made affect the file in real time. 

@soumyasmruti @ankushdesai @aman-goel

### CODE REVIEW

**Comment by aman-goel** [comment]
As the very first version, this looks to me good to be merged into master.

One request is to change the name of p.YAML-tmLanguage file to p.tmLanguage.yaml.

**Comment by soumyasmruti** [comment]
1. When you squash your commit provide a proper commit message instead of just "squashing commits".
2. Having screenshots for the output is great but add little more details in the description. Follow the following format. 

```
Section 1: Summary - Purpose of the change and the context (What, why and how)

Section 2: Testing - details of the testing done 
```

**Comment by ankushdesai** [comment]
Looks good to me!

File: syntaxes/p.YAML-tmLanguage [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+  captures:
+    "1":
+      name: keyword.statement.structural
+- name: meta.typedecl
+  match: (\btype\b)\s*([\w'?]+\b(?!<))?\s*
+  captures:
+    "1":
+      name: keyword.statement.structural
+    "2":
+      name: storage.type.other
```
Comment (soumyasmruti) [inline syntaxes/p.YAML-tmLanguage:22]: add new line after this statement and try to separate different sections using new line characters.

File: syntaxes/p.YAML-tmLanguage [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+  match: \b([1-9]+[0-9]*|0|[0-9]*\.[0-9]*)\b
+- name: constant.language #strings
+  match: (".*")
+
+- name: keyword.primitive_expressions
+  match: \$
+  captures:
+    '1': {name: keyword.primitive_expressions}
+
+repository:
```
Comment (soumyasmruti) [inline syntaxes/p.YAML-tmLanguage:52]: add more descriptive comments on the meaning of this section that. will help to understand the code better.

**Review by soumyasmruti — COMMENTED** [review]
Good work on the initial PR. Please take a look at comments and update the review.

**Comment by suesther** [comment]
> As the very first version, this looks to me good to be merged into master.
> 
> One request is to change the name of p.YAML-tmLanguage file to p.tmLanguage.yaml.

Unfortunately, if I do this, I am unable to convert the yaml file into the json tmLanguage file needed to run the extension!

File: syntaxes/p.YAML-tmLanguage (hunk 1 above)
Comment (aman-goel) [reply syntaxes/p.YAML-tmLanguage:22]: @suesther 
Reminder! Let's resolve @soumyasmruti's comments and merge this PR.

**Review by aman-goel — APPROVED** [review]

**Pull request merged**